streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -698,6 +707,9 @@ private long processAndPunctuateStreamTime(final Map<TaskId, StreamTask> tasks,
         } while (totalProcessedEachRound != 0);
 
         // go over the tasks again to punctuate or commit
+        // TODO: the thrown exception could be ProducerFenced, and hence need to be handled on the task that throws it
```

Comment:
Yup the TODOs are just for reviewers to see if they make sense as a follow-up PR (since it is likely that someone reviewing this will also review the follow-up PR or work on them as well). So it is just an FYI.

I will remove all these TODOs now if you have made a pass and agrees on these comments.